<?xml version="1.0" ?>
<launch>
    <arg name="mesh_path" value="$(find jaco2_surface_model)/jaco2_surface_meshes/"/>

    <group ns="muse_armcl">
        <!-- update models -->

        <!-- prediction model -->

        <!-- map providers -->
        <group ns="mesh_map">
            <param name="class"      value="muse_armcl::MeshMapLoader" />
            <param name="base_class" value="muse_armcl::MeshMapProvider" />
            <param name="path"       value="$(arg mesh_path)" />
        </group>

        <!-- data providers -->

        <!-- sampling algorithms -->
        <group ns="uniform_sampler">
        </group>

        <group ns="normal_sampler">
        </group>

        <group ns="resampling">
        </group>

        <!-- density estimation -->
        <group ns="density">
        </group>

        <!-- scheduling -->
        <group ns="scheduler">
        </group>

        <!-- particle filter setup -->
        <group ns="particle_filter">
            <param name="world_frame"                   value="/world"/>
            <param name="reset_weights_after_insertion" value="true"  />
            <param name="reset_weights_to_one"          value="false" />
            <param name="enable_lag_correction"         value="false" />

            <!--<param name="sample_size" value=""/>-->       <!-- if set, maximum and minimum are this value -->
            <param name="minimum_sample_size" value="100" />  <!-- if not sample_size is already set -->
            <param name="maximum_sample_size" value="5000"/>  <!-- if not sample_size is already set -->

        </group>
    </group>

    <!-- MuSe ARMCL node -->
    <node type="muse_armcl_node" name="muse_armcl" pkg="muse_armcl" output="screen" clear_params="true" />
</launch>
